diag_manager/fms_diag_file_object.F90

```diff
@@ -1119,6 +1120,40 @@ subroutine write_time_metadata(this)
 
 end subroutine write_time_metadata
 
+!> \brief Write out the field data to the file
+subroutine write_field_data(this, field_obj, buffer_obj)
+  class(fmsDiagFileContainer_type), intent(in), target   :: this !< The file object
+  type(fmsDiagField_type),                 intent(in), target :: field_obj(:)
+  type(fmsDiagOutputBufferContainer_type), intent(in), target :: buffer_obj(:)
+
+  class(fmsDiagFile_type), pointer     :: diag_file      !< Diag_file object to open
+  class(FmsNetcdfFile_t),  pointer     :: fileobj        !< Fileobj to write to
```

Comment:
OK, so I know who came up with this variable name (I deserve the blame), but I think calling it `fms2io_obj` or `fms2io_file` is a better name.  In this version of the diag manager, we have a file object for the diag manager and for fms2_io.  `fileobj` is somewhat confusing here.  We can change this later if you want.

Reply:
@mcallic2 is changing this later